ontology-engine/graph-engine_2.11/src/main/scala/org/sunbird/graph/schema/DefinitionNode.scala

```diff
@@ -20,49 +20,50 @@ object DefinitionNode {
       val graphId: String = request.getContext.get("graph_id").asInstanceOf[String]
       val version: String = request.getContext.get("version").asInstanceOf[String]
       val schemaName: String = request.getContext.get("schemaName").asInstanceOf[String]
-      val categoryId: String = getPrimaryCategory(request.getRequest, schemaName, request.getContext.getOrDefault("channel", "all").asInstanceOf[String])
-      val definition = DefinitionFactory.getDefinition(graphId, schemaName, version, categoryId)
+      val objectCategoryDefinition: ObjectCategoryDefinition = getObjectCategoryDefinition(request.getRequest.getOrDefault("primaryCategory", "").asInstanceOf[String],
+        schemaName, request.getContext.getOrDefault("channel", "all").asInstanceOf[String])
+      val definition = DefinitionFactory.getDefinition(graphId, schemaName, version, objectCategoryDefinition)
       definition.validateRequest(request)
       val inputNode = definition.getNode(request.getRequest)
 	  updateRelationMetadata(inputNode)
       definition.validate(inputNode, "create", setDefaultValue) recoverWith { case e: CompletionException => throw e.getCause}
   }
 
-    def getExternalProps(graphId: String, version: String, schemaName: String, categoryId: String = "")(implicit ec: ExecutionContext, oec: OntologyEngineContext): List[String] = {
-        val definition = DefinitionFactory.getDefinition(graphId, schemaName, version, categoryId)
+    def getExternalProps(graphId: String, version: String, schemaName: String, ocd: ObjectCategoryDefinition = new ObjectCategoryDefinition())(implicit ec: ExecutionContext, oec: OntologyEngineContext): List[String] = {
```

Comment:
```suggestion
    def getExternalProps(graphId: String, version: String, schemaName: String, ocd: ObjectCategoryDefinition =  ObjectCategoryDefinition())(implicit ec: ExecutionContext, oec: OntologyEngineContext): List[String] = {
```

`new` is not required for creating an object from case class (ObjectCategoryDefinition). Please check it once.